pkg/secrets/1password.go

```diff
@@ -0,0 +1,125 @@
+package secrets
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/1password/onepassword-sdk-go"
+)
+
+//go:generate mockgen -destination=mocks/mock_onepassword.go -package=mocks -source=1password.go OPSecretsService,OPClient
+
+// OPSecretsService defines the interface for the 1Password Secrets service
+type OPSecretsService interface {
+	Resolve(ctx context.Context, secretReference string) (string, error)
+}
+
+// OPClient defines the interface for the 1Password client
+type OPClient interface {
+	Secrets() OPSecretsService
+}
+
+// onePasswordAdapter adapts the onepassword-sdk-go client to our OPClient interface
+type onePasswordAdapter struct {
+	client *onepassword.Client
+}
+
+func (a *onePasswordAdapter) Secrets() OPSecretsService {
+	return &secretsServiceAdapter{client: a.client}
+}
+
+// secretsServiceAdapter adapts the onepassword-sdk-go secrets service to our OPSecretsService interface
+type secretsServiceAdapter struct {
+	client *onepassword.Client
+}
+
+func (s *secretsServiceAdapter) Resolve(ctx context.Context, secretReference string) (string, error) {
+	return s.client.Secrets().Resolve(ctx, secretReference)
+}
+
+// OnePasswordManager manages secrets in 1Password.
+type OnePasswordManager struct {
+	client   *onepassword.Client
+	opClient OPClient // Interface for easier mocking
+}
+
+// GetSecret retrieves a secret from the secret store.
+func (opm *OnePasswordManager) GetSecret(path string) (string, error) {
+	if !strings.Contains(path, "op://") {
+		return "", fmt.Errorf("invalid secret path: %s", path)
+	}
+
+	// Use the opClient interface if available, otherwise use the direct client
+	var secret string
+	var err error
+	if opm.opClient != nil {
+		secret, err = opm.opClient.Secrets().Resolve(context.TODO(), path)
+	} else {
+		secret, err = opm.client.Secrets().Resolve(context.TODO(), path)
+	}
+
+	if err != nil {
+		return "", fmt.Errorf("error resolving secret: %v", err)
+	}
+
+	return secret, nil
+}
+
+// SetSecret is not supported for 1Password.
+// We will support this if we notice an increase in demand.
+func (opm *OnePasswordManager) SetSecret(name, value string) error {
+	return nil
+}
+
+// DeleteSecret is not supported for 1Password.
+// We will support this if we notice an increase in demand.
+func (opm *OnePasswordManager) DeleteSecret(name string) error {
+	return nil
+}
+
+// ListSecrets is not supported for 1Password.
+// We will support this if we notice an increase in demand.
+func (opm *OnePasswordManager) ListSecrets() ([]string, error) {
+	return nil, nil
+}
+
+// Cleanup is not supported or needed for 1Password.
+func (opm *OnePasswordManager) Cleanup() error {
+	return nil
+}
+
+// NewOnePasswordManager creates an instance of OnePasswordManager.
+func NewOnePasswordManager() (Provider, error) {
+	token := os.Getenv("OP_SERVICE_ACCOUNT_TOKEN")
```

Comment:
Let's add docs for this environment variable in the README. Let's also add this as a constant on the top of the file.

Reply:
The `OP_SERVICE_ACCOUNT_TOKEN` string itself or the token?